src/dune_rules/cxx_flags.mli

```diff
@@ -3,10 +3,6 @@
 
 open Import
 
-type phase =
-  | Compile
-  | Link
```

Comment:
Is this refactoring essential to the fix?

Reply:
that type isn't used anywhere anymore and the removal of its use is essential to the fix (get_compile_flags is the only one that need the extra argument and places where get_link_flags is called do not have any way of getting that value, plus that would be wasteful)

Reply:
You could have also just modified the type to be:

```ocaml
type phase =
  | Compile of Ocaml_config.t
  | Link
```

That would be a bit closer the old API and preserve some of the call sites.

It's not essential in this case as the refactoring is trivial, but in general I would like fixes and refactors to be as separate as possible.